nautobot/dcim/models/devices.py

```diff
@@ -955,6 +956,55 @@ def get_children(self):
         """
         return Device.objects.filter(parent_bay__device=self.pk)
 
+    @property
+    def modules(self):
+        """
+        Return all child Modules installed in ModuleBays within this Device.
+        """
+        # Supports Device->ModuleBay->Module->ModuleBay->Module->ModuleBay->Module->ModuleBay->Module
+        # This query looks for modules that are installed in a module_bay and attached to this device
+        # We artificially limit the recursion to 4 levels or we would be stuck in an infinite loop.
+        recursion_depth = 4
+        qs = Module.objects.all()
+        query = models.Q()
+        for level in range(recursion_depth):
+            recursive_query = "parent_module_bay__parent_module__" * level
+            query = query | models.Q(**{f"{recursive_query}parent_module_bay__parent_device": self})
+        return qs.filter(query)
+
+    @property
+    def all_console_ports(self):
```

Comment:
Let's add docstrings to these please. Also might be worth a sentence or two in the Device documentation?